### DISCUSSION TITLE
no keyboard or mouse on client after server update

### DISCUSSION THREAD

**Discussion by dimrim** [ltsp/ltsp#887] [General]
Hello, can someone maybe help us?

We use an Ltsp-server, version 23.02-1+deb12u1, from the stable Debian repositories. The current Debian software, 64-bit, "Bookworm" is installed on the server. The clients are chrootless.

After the last debian security update:

    • base-files (version 12.4+deb12u2) is updated to version 12.4+deb12u4
    • distro-info-data (version 0.58) is updated to version 0.58+deb12u1
    • exfatprogs (version 1.2.0-1) is updated to version 1.2.0-1+deb12u1
    • libde265-0 (version 1.0.11-1) is updated to version 1.0.11-1+deb12u1
      
    • exim4-base (version 4.96-15+deb12u2) is updated to version 4.96-15+deb12u3
    • exim4-config (version 4.96-15+deb12u2) is updated to version 4.96-15+deb12u3
    • exim4-daemon-light (version 4.96-15+deb12u2) is updated to version 4.96-15+deb12u3
    • gir1.2-javascriptcoregtk-4.0 (version 2.42.2-1~deb12u1) is updated to version 2.42.3-1~deb12u1
    • gir1.2-webkit2-4.0 (version 2.42.2-1~deb12u1) is updated to version 2.42.3-1~deb12u1
    • gstreamer1.0-plugins-bad (version 1.22.0-4+deb12u2) is updated to version 1.22.0-4+deb12u3
    • libgnutls-dane0 (version 3.7.9-2) is updated to version 3.7.9-2+deb12u1
    • libgnutls30 (version 3.7.9-2) is updated to version 3.7.9-2+deb12u1
    • libgstreamer-plugins-bad1.0-0 (version 1.22.0-4+deb12u2) is updated to version 1.22.0-4+deb12u3
    • libjavascriptcoregtk-4.0-18 (version 2.42.2-1~deb12u1) is updated to version 2.42.3-1~deb12u1
    • libjavascriptcoregtk-4.1-0 (version 2.42.2-1~deb12u1) is updated to version 2.42.3-1~deb12u1
    • libmateweather-common (version 1.26.0-1.1) is updated to version 1.26.0-1.1+deb12u1
    • libmateweather1 (version 1.26.0-1.1) is updated to version 1.26.0-1.1+deb12u1
    • libminizip1 (version 1.1-8+b1) is updated to version 1.1-8+deb12u1
    • libnghttp2-14 (version 1.52.0-1) is updated to version 1.52.0-1+deb12u1
    • libqpdf29 (version 11.3.0-1) is updated to version 11.3.0-1+deb12u1
    • libtiff6 (version 4.5.0-6) is updated to version 4.5.0-6+deb12u1
    • libwebkit2gtk-4.0-37 (version 2.42.2-1~deb12u1) is updated to version 2.42.3-1~deb12u1
    • libwebkit2gtk-4.1-0 (version 2.42.2-1~deb12u1) is updated to version 2.42.3-1~deb12u1
    • tzdata (version 2023c-5) is updated to version 2023c-5+deb12u1
      
    • libnss-myhostname (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1
    • libnss-resolve (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1
    • libnss-systemd (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1
    • libpam-systemd (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1
    • libsystemd-shared (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1
    • libsystemd0 (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1
    • libudev1 (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1
    • systemd (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1
    • systemd-resolved (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1
    • systemd-sysv (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1
    • systemd-timesyncd (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1
    • udev (version 252.17-1~deb12u1) is updated to version 252.19-1~deb12u1

    • xserver-common (version 2:21.1.7-3+deb12u2) is updated to version 2:21.1.7-3+deb12u3
    • xserver-xorg-core (version 2:21.1.7-3+deb12u2) is updated to version 2:21.1.7-3+deb12u3
    • xserver-xorg-legacy (version 2:21.1.7-3+deb12u2) is updated to version 2:21.1.7-3+deb12u3

Now after our clients boot, the mouse indicator doesn’t move and the keyboard doesn't respond either. Anyway the clients can be operated via epoptes. I suspect updating udev is causing the problem. But there is no possibility of a downgrade and I cannot test it.

Maybe I'm completely wrong. Can anyone please give me some advice?

Thanks

**Comment by alkisg** [comment]
Boot a client, open Epoptes, right click open a root terminal, and run:

    uname -r
    du -sh /lib/modules/*

Then check if the booted kernel modules dir actually has all the modules.

**Reply by dimrim** [reply]
Thank you for the fast feedback!
I think the booted kernel modules dir has all the modules:

uname -r
du -sh /lib/modules/*
6.1.0-12-amd64
0 /lib/modules/5.10.0-23-amd64
0 /lib/modules/5.10.0-25-amd64
385M /lib/modules/6.1.0-13-amd64
385M /lib/modules/6.1.0-15-amd64

ls 
/lib/modules/5.10.0-23-amd64
/lib/modules/5.10.0-25-amd64
/lib/modules/6.1.0-13-amd64
/lib/modules/6.1.0-15-amd64

Or what do you mean?

**Reply by dimrim** [reply]
In any case, the contents of the ./6.1.0-15-amd64 and ./6.1.0-13-amd64 folders are almost the same. Only a few things were added in ./6.1.0-15-amd64.

**Comment by alkisg** [comment]
The problem is that you are booted with kernel 6.1.0-12-amd64 while no /lib/modules/6.1.0-12-amd64 directory exists.
I.e. there are no modules at all for this kernel, so no modules for mouse, keyboard and a lot of other stuff.

Try running `sudo ltsp kernel`, then reboot the client and see if it's then booted with 6.1.0-15-amd64.

**Reply by dimrim** [reply]
Thanks alot! This solved the problem:

sudo ltsp image /,,/boot,subdir=boot
sudo lspt kernel
sudo ltsp initrd

Sometimes we're just blind.

**Reply by dimrim** [reply]
I haven't found where I can mark the discussion as answered. Thank you again.
